lib/Doctrine/MongoDB/Event/MutableEventArgs.php

```diff
@@ -54,4 +56,32 @@ public function isDataChanged()
     {
         return $this->isDataChanged;
     }
+
+    /**
+     * @since 1.3
+     * @return array
+     */
+    public function getOptions()
+    {
+        return $this->isOptionsChanged ? $this->changedOptions : parent::getOptions();
```

Comment:
I agree that the wording is a bit odd, as it was copied from `$changedData`, which is singular. The intention here is definitely to just track whether the array itself has changed in some way and always return the current state. We recalc `$isOptionsChanged` on each set since we can only access the parent class' options through the getter and have no API to modify it.